packages/backend/src/managers/llama-stack/llamaStackManager.ts

```diff
@@ -0,0 +1,251 @@
+    this.podmanConnection.onPodmanConnectionEvent(this.watchMachineEvent.bind(this));
+    this.containerRegistry.onStartContainerEvent(this.onStartContainerEvent.bind(this));
+    this.containerRegistry.onStopContainerEvent(this.onStopContainerEvent.bind(this));
+  }
+
+  private async refreshLlamaStackContainer(id?: string): Promise<void> {
+    const containers = await containerEngine.listContainers();
+    const containerInfo = containers
+      .filter(c => !id || c.Id === id)
+      .filter(c => c.State === 'running' && c.Labels && LLAMA_STACK_CONTAINER_LABEL in c.Labels)
+      .at(0);
+    if ((id && containerInfo) || !id) {
+      this.#containerInfo = containerInfo
+        ? {
+            containerId: containerInfo.Id,
+            port: parseInt(containerInfo.Labels[LLAMA_STACK_API_PORT_LABEL]),
+          }
+        : undefined;
+    }
+  }
+
+  private async watchMachineEvent(event: PodmanConnectionEvent): Promise<void> {
+    if ((event.status === 'started' && !this.#containerInfo) || (event.status === 'stopped' && this.#containerInfo)) {
+      await this.refreshLlamaStackContainer();
+    }
+  }
+
+  private async onStartContainerEvent(event: ContainerEvent): Promise<void> {
+    await this.refreshLlamaStackContainer(event.id);
+  }
+
+  private onStopContainerEvent(event: ContainerEvent): void {
+    console.log('event id:', event.id, ' containerId: ', this.#containerInfo?.containerId);
+    if (this.#containerInfo?.containerId === event.id) {
+      this.#containerInfo = undefined;
+      this.taskRegistry.deleteByLabels({ trackingId: LLAMA_STACK_CONTAINER_TRACKINGID });
+    }
+  }
+
+  async getLlamaStackContainer(): Promise<LlamaStackContainerInfo | undefined> {
+    if (!this.#initialized) {
+      const containers = await containerEngine.listContainers();
+      this.#containerInfo = containers
+        .filter(c => c.State === 'running' && c.Labels && LLAMA_STACK_CONTAINER_LABEL in c.Labels)
+        .map(c => ({
+          containerId: c.Id,
+          port: parseInt(c.Labels[LLAMA_STACK_API_PORT_LABEL]),
+        }))
+        .at(0);
+      this.#initialized = true;
+    }
+    return this.#containerInfo;
+  }
+
+  async requestCreateLlamaStackContainer(config: LlamaStackContainerConfiguration): Promise<void> {
+    // create a tracking id to put in the labels
+    const trackingId: string = LLAMA_STACK_CONTAINER_TRACKINGID;
+
+    const labels = {
+      trackingId: trackingId,
+    };
+
+    const task = this.taskRegistry.createTask('Creating Llama Stack container', 'loading', {
+      trackingId: trackingId,
+    });
+
+    let connection: ContainerProviderConnection | undefined;
+    if (config.connection) {
+      connection = this.podmanConnection.getContainerProviderConnection(config.connection);
+    } else {
+      connection = this.podmanConnection.findRunningContainerProviderConnection();
+    }
+
+    if (!connection) throw new Error('cannot find running container provider connection');
```

Comment:
feels like this should be above the createTask, as this does not make the task fails, and just make the all process running forever